tests/version_test.go

```diff
@@ -0,0 +1,64 @@
+package tests
+
+import (
+	"context"
+	"testing"
+
+	"github.com/arangodb/go-driver"
+	api "github.com/arangodb/kube-arangodb/pkg/apis/deployment/v1alpha"
+	"github.com/arangodb/kube-arangodb/pkg/client"
+	"github.com/arangodb/kube-arangodb/pkg/util"
+	"github.com/arangodb/kube-arangodb/pkg/util/arangod"
+	"github.com/dchest/uniuri"
+)
+
+func TestIsVersionSet(t *testing.T) {
+	longOrSkip(t)
+	c := client.MustNewInCluster()
+	kubecli := mustNewKubeClient(t)
+	ns := getNamespace(t)
+	expectedVersion := "3.3.17"
+	// Prepare deployment config
+	depl := newDeployment("test-auth-sng-def-" + uniuri.NewLen(4))
+	depl.Spec.Mode = api.NewMode(api.DeploymentModeSingle)
+	depl.Spec.SetDefaults(depl.GetName())
+	depl.Spec.Image = util.NewString("arangodb/arangodb:" + expectedVersion)
+	// Create deployment
+	apiObject, err := c.DatabaseV1alpha().ArangoDeployments(ns).Create(depl)
+	if err != nil {
+		t.Fatalf("Create deployment failed: %v", err)
+	}
+	defer deferedCleanupDeployment(c, depl.GetName(), ns)
+	// Wait for deployment to be ready
+	depl, err = waitUntilDeployment(c, depl.GetName(), ns, deploymentIsReady())
+	if err != nil {
+		t.Fatalf("Deployment not running in time: %v", err)
+	}
+
+	single := depl.Status.Members.Single
+	if single == nil || len(single) == 0 {
+		t.Fatalf("single member is empty")
+	}
+
+	if string(single[0].ArangoVersion) != expectedVersion {
```

Comment:
I think we have special functions for comparing versions. Please have a look at the package containing the type of `ArangoVersion`.